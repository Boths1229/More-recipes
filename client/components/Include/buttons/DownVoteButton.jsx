import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description functional component to render downvote button
 *
 * @param {object} props
 *
 * @returns {JSX} return JSX
 */
const DownVoteButton = (props) => {
  const { downvoteRecipe, recipeId, downvotes } = props;
  return (
    <div>
      <button className="btn btn-orange mr-2" type="button">
        <i
          className="fa fa-thumbs-down fa-2x btn-white"
          aria-hidden="true"
          onClick={() => downvoteRecipe(recipeId)}
        />
      </button>
      <span className="mr-2">
        {downvotes}
      </span>
    </div>
  );
};

DownVoteButton.propTypes = {
  downvoteRecipe: PropTypes.func.isRequired,
  recipeId: PropTypes.number.isRequired,
  downvotes: PropTypes.number.isRequired
};

export default DownVoteButton;
