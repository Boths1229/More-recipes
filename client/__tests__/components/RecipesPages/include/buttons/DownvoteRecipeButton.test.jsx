import React from 'react';
import { shallow } from 'enzyme';
import DownVoteButton from '../../../../../components/Include/buttons/DownVoteButton';

describe('Downvote recipe button component', () => {
  it('matches snapshot', () => {
    const output = shallow(<DownVoteButton />);
    expect(output).toMatchSnapshot();
  });
  it('calls onClick', () => {
    const props = {
      downvoteRecipe: jest.fn()
    };
    const output = shallow(<DownVoteButton {...props} />);
    const button = output.find('i[aria-hidden="true"]');
    button.simulate('click');
    expect(props.downvoteRecipe).toHaveBeenCalled();
  });
});
