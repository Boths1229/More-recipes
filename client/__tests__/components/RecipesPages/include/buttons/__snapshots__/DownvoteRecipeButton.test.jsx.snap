// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Downvote recipe button component matches snapshot 1`] = `
<div>
  <button
    className="btn btn-orange mr-2 down favorite-btn"
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-thumbs-down fa-2x btn-white"
      onClick={[Function]}
    />
  </button>
  <span
    className="mr-2"
  />
</div>
`;
